pymeasure/instruments/rigol/rigol_ds1000z.py

```diff
@@ -0,0 +1,849 @@
+        validator=strict_range,
+        values=[2, 1024],
+        cast=int,
+    )
+
+    sample_rate = Instrument.measurement(
+        ":ACQ:SRAT?",
+        "Measure the current sample rate in samples per second (float).",
+        cast=float,
+    )
+
+    @property
+    def memory_depth(self) -> Union[int, str]:
+        """Control the acquisition memory depth in points or 'AUTO'.
+
+        The allowable discrete values depend on how many analogue and digital
+        channels are currently enabled. The driver counts the active analogue
+        inputs and uses :meth:`set_digital_channel_hint` as an optional cue for
+        the digital pod state before applying validation.
+        """
+
+        reply = self.ask(":ACQ:MDEP?")
+        return _parse_memory_depth(reply)
+
+    @memory_depth.setter
+    def memory_depth(self, value: Union[str, int]) -> None:
+        target, numeric = _normalize_memory_depth_value(value)
+        if numeric is not None:
+            allowed = self._allowed_memory_depth_values()
+            if numeric not in allowed:
+                raise ValueError(
+                    f"Memory depth {numeric} pts is not permitted for the current "
+                    f"channel configuration. Allowed values: {sorted(allowed)} or 'AUTO'."
+                )
+        self.write(f":ACQ:MDEP {target}")
+
+    trigger_source = Instrument.control(
+        ":TRIG:EDGE:SOUR?",
+        ":TRIG:EDGE:SOUR %s",
+        "Control the trigger source selection (str).",
+        validator=strict_discrete_set,
+        values=_WAVEFORM_SOURCES,
+    )
+
+    trigger_slope = Instrument.control(
+        ":TRIG:EDGE:SLOP?",
+        ":TRIG:EDGE:SLOP %s",
+        "Control the trigger slope (str): 'POS' or 'NEG'.",
+        validator=strict_discrete_set,
+        values=["POS", "NEG"],
+    )
+
+    trigger_coupling = Instrument.control(
+        ":TRIG:COUP?",
+        ":TRIG:COUP %s",
+        "Control the trigger coupling mode (str).",
+        validator=strict_discrete_set,
+        values=["AC", "DC", "HFRej", "LFRej", "NOIS"],
+    )
+
+    trigger_sweep = Instrument.control(
+        ":TRIG:SWE?",
+        ":TRIG:SWE %s",
+        "Control the trigger sweep mode (str): 'AUTO', 'NORM', or 'SING'.",
+        validator=strict_discrete_set,
+        values=["AUTO", "NORM", "SING"],
+    )
+
+    trigger_level = Instrument.control(
+        ":TRIG:LEV?",
+        ":TRIG:LEV %f",
+        "Control the trigger level in volts (float).",
+        cast=float,
+    )
+
+    waveform_source = Instrument.control(
+        ":WAV:SOUR?",
+        ":WAV:SOUR %s",
+        "Control the waveform source used for data transfers (str).",
+        validator=strict_discrete_set,
+        values=_WAVEFORM_SOURCES,
+    )
+
+    waveform_mode = Instrument.control(
+        ":WAV:MODE?",
+        ":WAV:MODE %s",
+        "Control the waveform acquisition mode for transfer (str).",
```

Comment:
A few hints to the use of enums:
1. you should state it in the docstring. If you sue something like the code below, it will link to the class in the API documenation
2. You can use the enum as `values`, as the validator will verify, whether the value is `in` values.
3. You can use also `get_process=lambda val: WaveformMode(val)` in order to return a WaveformMode instance!


```suggestion
        "Control the waveform acquisition mode for transfer (:class:`WaveformMode`).",
```